src/datasets/features/features.py

```diff
@@ -1360,7 +1378,11 @@ def generate_from_dict(obj: Any):
     if "_type" not in obj or isinstance(obj["_type"], dict):
         return {key: generate_from_dict(value) for key, value in obj.items()}
     obj = dict(obj)
-    class_type = globals()[obj.pop("_type")]
+    _type = obj.pop("_type")
+    class_type = _FEATURE_TYPES.get(_type, None) or globals().get(_type, None)
+
+    if class_type is None:
+        raise ValueError(f"Feature type '{_type}' not found. Available feature types: {list(_FEATURE_TYPES.keys())}")
```

Comment:
Maybe populate _FEATURE_TYPES with the existing features so that the error message doesn't show an empty list by default ?

Reply:
Yeah, I had `{k: v for k, v in globals().items() if isinstance(v, FeatureType)}` to avoid hard coding it. Subscripted don't work in 3.8 unfortunately. I'll just hard code it since it's experimental anyways.